### ISSUE TITLE
[Discussion] ECIP-1101: Protocol Compact 2020

### ISSUE THREAD

**Issue by TokenHash** [ethereumclassic/ECIPs#386]
Discussion for ECIP-1101: Protocol Compact 2020.

**Comment by antsankov** [comment]
Wholeheartedly agree with all of this. A reasonable proto-treasury, keccak-256 algorithm dominance, and MESS protocol will absolutely turbocharge the ETC chain. This roadmap is good to go and should be implemented. 

ECIP-1049 has been updated to be in line with the spirit and urgency of the Protocol Compact:

Activation Block: 12,000,000 (approx. 4 months from acceptance - January 2021)
Fallback Activation Block: 12,500,000 (approx. 7 months from acceptance - April 2021)
If not activated by Block 12,500,000, ECIP-1049 is voided and moved to Rejected.

**Comment by gitr0n1n** [comment]
ECIP-1049 is unnecessary to address 51% attacks. Sounds like a great roadmap for a new chain/fork though.

**Comment by q9f** [comment]
What's the purpose of this document? And how do we benefit from this bundle? What role does EIP-1099 play?

And - this is my personal opinion - why do you believe - out of all proposals - that we need on-chain governance to make Ethereum Classic progress in future? We have proven that Classic is alive by planning and executing 3 protocol upgrades in the last 12 months and we are about to finalize the 4th. How would locking ETC and voting on chain help with progress? I personally believe this is just distraction. Also, do we know who the main ETC holders are?

**Comment by TokenHash** [comment]
@q9f 

> What's the purpose of this document? And how do we benefit from this bundle?

Noticing there is considerable support for changes I don't agree with (treasury & MESS), and delays or less support for a change I agree with (Keccak256), but at the same time ETC needs a fast solution ASAP as it is losing and is under the threat of losing further liquidity, a key component of its security model, the purpose of this ECIP-1101 is to offer the acceptance of treasury and MESS with time limits (5 years) in exchange for the permanent implementation of Keccak256. It also seeks to make these decisions as soon as possible, without undermining the ECIP process in general.

> What role does EIP-1099 play?

ECIP-1101 implicitly discards 1099. The rationale is for ETC to move directly to SHA3 as it is a proven function, clean and easy to build hardware for, and relatively unique in the blockchain industry. It is my view that it is waste of time to continue trying to fix ETHash/ETChash.

> ...why do you believe - out of all proposals - that we need on-chain governance to make Ethereum Classic progress in future? We have proven that Classic is alive by planning and executing 3 protocol upgrades in the last 12 months and we are about to finalize the 4th. How would locking ETC and voting on chain help with progress? I personally believe this is just distraction. Also, do we know who the main ETC holders are that would rule the chain in future?

I agree with you. I don't agree with the treasury, if that is what you are referring to by "on-chain governance". I think others like the treasury because it removes central power from ETC Labs, who are the only ones who have and are investing money in the system.

--

In summary I don't like the treasury nor MESS, but would like to see sha3 implemented ASAP and MESS and the treasury limited in time, so ETC eventually falls back to its original design.

If the above does not happen then I will oppose both MESS, the Treasury, and ECIP-1099.

**Comment by q9f** [comment]
The problem is SHA is, we should have started working towards an implementation last year. Now after the attacks, we kind of "woke up" and try to rush it. This change is probably the most involved that ETC will ever see and proposing a 9 months implementation timeline is the fastest possible.

Now is September and now work on the clients has been started yet. 9 Months is June 2021. In that way, 1049 is not our 51% problem solver and therefore, 1099 provides a first step towards an own mining algorithm and 1100 are training wheels we can activate today until we do not need them anymore (i.e., on 1049 activation date).

**Comment by TokenHash** [comment]
@q9f so what you are describing is also a sort of programme:

1. Now: implement ECIP-1099 + MESS
2. In nine months: implement ECIP-1049 + delete ETCash (1099) + MESS

I see the "training wheels" purpose as current hash power is so low that it doesn't make ETC stable anyway. But,

- why is ECIP-1099 ready to go and ECIP-1049 not? Isn't 1099 equally untested and immature?
- how can it be guaranteed that MESS is deleted? isn't it better to implement it with a termination date?

**Comment by TokenHash** [comment]
In any case, to implement MESS and 1099 now to wait for Keccak256 in 9 months, it's better to just implement MESS (with an expiration date) and wait for Keccak256 directly.

I see ECIP-1099 doesn't even solve the DAG growth. 

MESS at least minimizes attacks with any underlying algo, including the current ETHASH, while we wait for SHA3.

**Comment by q9f** [comment]
> @q9f so what you are describing is also a sort of programme:
> 
>     1. Now: implement ECIP-1099 + MESS
>     2. In nine months: implement ECIP-1049 + delete ETCash (1099) and delete MESS

Exactly. Both 1099 and 1100 are temporary solutions. EtcHash will be replaced by Keccak once this is ready. This buys us time to properly implement 1049.

> I see the "training wheels" purpose as current hash power is so low that it doesn't make ETC stable anyway. But,
> 
>     * why is ECIP-1099 ready to go and ECIP-1049 not? Isn't 1099 equally untested and immature?
>     * how can it be guaranteed that MESS is deleted? isn't it better to implement it with a termination date?

Because 1099 is literally just adding some lines to Ethash. It's a hotfix. It's neither elegant nor beautiful. But given that it is easy to implement and quick to activate, it buys us a lot of needed time to go through an involved change such as Keccak.

MESS is not a consensus feature. You can just turn it on and off as you like. For core-geth, it's something like `--ecbp1100 999999999` to disable it anytime. It does not even require developer intervention. On the other hand, if you want to help exchanges to protect their funds in times like that, they could coordinate and override it already by activating it on mainnet, e.g. `geth --classic --ecbp1100 11380000`.

**Comment by TokenHash** [comment]
> Exactly. Both 1099 and 1100 are temporary solutions. EtcHash will be replaced by Keccak once this is ready. This buys us time to properly implement 1049.

This is good news from my standpoint. I had the impression there was a pushback against Keccak256.

> I would say 12 months would be a safe timeline, 9 months is the absolute minimum.

Based on your description above, 9-12 months makes sense. I agree with that.

> Keccak does not requires us to wait 9 months, it requires us to work hard 9 months.

Yes. I used the term "wait" but I understand it involves real work from the part of devs participating in the change. For non-devs it's just "wait" that's why I used that term.

--

So, in terms of MESS and 1099 as hot fixes to eventually migrate to 1049 I have that clear now. 

As to MESS, I don't think it is a good idea to have it as an option for some or other exchanges and miners to decide independently  whether to follow one chain or the other, that may lead to splits no? In any case I would do a hard fork with MESS and Epoch Duration in one hard fork, and in parallel work on Keccak256 for 12 months. (again if MESS solves reorgs, I don't know why go through the work of doing Epoch Duration)

As to the Treasury, that leaves that issue as a divisive item. From my standpoint, The MESS -> Epoch Duration -> Keccak256 track is acceptable.

**Comment by developerkevin** [comment]
I think some of the main takeaways to this **informational ECIP** is that these "training wheel" proposals should have some point of removal specified in their ECIP. Otherwise they could theoretically be there on nodes forever, just unused, even after hash rate hopefully returns and the network returns to normal.

I understand that MESS and others have certain points that must be met in order for them to 1. turn  on and 2. turn off but I think putting a time limit on them with an option to renew is a better way forward.

**Comment by q9f** [comment]
For 1099 we could clearly specify the 1049 transition as definite replacement.

For 1100 we could specify either a target hashrate, mabe 20 TH/s, and in case this is not met, also set the 1049 transition as target, maybe not the same block, but like 4-6 weeks after keccak we can safely turn it off.

**Comment by TokenHash** [comment]
Closing PR. Closing discussion.
